dashboard/src/components/AppUpgrade/AppUpgrade.tsx

```diff
@@ -75,7 +75,10 @@ function AppUpgrade() {
     dispatch(
       actions.charts.getDeployedChartVersion(
         {
-          context: { cluster: cluster, namespace: repoNamespace ?? "" },
+          context: {
+            cluster: app?.availablePackageRef?.context?.cluster ?? cluster,
```

Comment:
Thanks Anthony. Yes, this should definitely be defaulting to the cluster of the available package, rather than the cluster of the installed app. +1 from me. Is the other discussion below resolved Antonio?

Reply:
Yep, everything is ready to go, I think. Indeed, given how messy the upgrade view was becoming I gave it a shot and try to perform some improvements (https://github.com/kubeapps/kubeapps/pull/3540).
However, the scope of the change is much bigger and I don't think we should release with my PR, but with this simpler and laser-focus one.

Let me jump in and merge the latest changes in the main branch (which hopefully should fix the CI tests...)

Reply:
@antgamdia that PR is def gonna be a good improvement. There are a lot of rerenders that end up going all the way down the chain and they _could_ be optimized and improve the performance of the page for sure

Reply:
> There are a lot of rerenders that end up going all the way down the chain

Yeah... in fact we have been facing some CI issues because of that (the dropdown was being re-rendered multiple times making our e2e engine chose the wrong version)

> Let me jump in and merge the latest changes in the main branch (which hopefully should fix the CI tests...)

I've tried but no permissions to edit this PR, I think. Please, @aanthonyrizzo, pull the latest changes in `master` and merge them into your branch. Let's see if tomorrow we can start tagging a release :P

Reply:
AH. Sorry we changed the ownership of the repo from my user to our organization. I just rebased so should be up to date.

Reply:
Mmm, perhaps it is why the CI is failing when building the images? Not sure, but maybe we should create another PR from this repo?

Reply:
Ah that could be. I'm going to be away for a bit but feel free to cherry pick the commits into another branch and merge :)

Reply:
Mmm, don't know, re-triggering it made the trick. CI things...
Merging! Thanks again for the contributions :)